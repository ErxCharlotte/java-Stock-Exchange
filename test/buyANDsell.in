ADD trader1 100.00
ADD trader2 20.00
IMPORT trader1 ABC 100
IMPORT trader1 CDE 1000
SELL trader1 ABC 10 -10
SELL trader1 ABC -1 100
SELL trader3 ABC 100 100
SELL trader1 LPL 100 100
SELL trader1 ABC 10000 100
SELL trader1 ABC 100 5
INVENTORY trader1
BUY trader2 LPL 1 -10
BUY trader2 LPL -100 30
BUY trader3 ABC 100 10
BUY trader2 CDE 100 10
BUY trader2 ABC 10 15
INVENTORY trader1
INVENTORY trader2
BALANCE trader1
BALANCE trader2
SELL trader1 CDE 500 5
INVENTORY trader1
INVENTORY trader2
BALANCE trader1
BALANCE trader2
AMOUNT trader1 CDE
BUY trader2 CDE 400 5
BUY trader2 CDE 100 10
SELL trader1 CDE 100 5
EXIT