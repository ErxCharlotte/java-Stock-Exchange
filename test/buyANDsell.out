$ Success.
$ Success.
$ Trader now has 100.00 units of ABC.
$ Trader now has 1000.00 units of CDE.
$ Order could not be placed onto the market.
$ Order could not be placed onto the market.
$ No such trader in the market.
$ Order could not be placed onto the market.
$ Order could not be placed onto the market.
$ No trades could be made, order added to sell book.
$ CDE
$ Order could not be placed onto the market.
$ Order could not be placed onto the market.
$ No such trader in the market.
$ No trades could be made, order added to buy book.
$ Product bought in entirety, trades as follows:
trader1->trader2: 10.00xABC for $5.00.
$ CDE
$ ABC
$ $150.00
$ $-30.00
$ Product sold in part, trades as follows:
trader1->trader2: 100.00xCDE for $10.00.
$ CDE
$ ABC
CDE
$ $1150.00
$ $-1030.00
$ 500.00
$ Product bought in entirety, trades as follows:
trader1->trader2: 400.00xCDE for $5.00.
$ No trades could be made, order added to buy book.
$ Product sold in entirety, trades as follows:
trader1->trader2: 100.00xCDE for $10.00.
$ Have a nice day.